fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java

```diff
@@ -0,0 +1,327 @@
+
+            for (int i = 1; i < relation.getRelations().size(); ++i) {
+                if (relation instanceof UnionRelation) {
+                    sqlBuilder.append(" UNION ");
+                } else if (relation instanceof ExceptRelation) {
+                    sqlBuilder.append(" EXCEPT ");
+                } else {
+                    sqlBuilder.append(" INTERSECT ");
+                }
+
+                sqlBuilder.append(relation.getQualifier() == SetQualifier.ALL ? "ALL " : "");
+
+                Relation setChildRelation = relation.getRelations().get(i);
+                if (setChildRelation instanceof SetOperationRelation) {
+                    sqlBuilder.append("(");
+                }
+                sqlBuilder.append(visit(setChildRelation));
+                if (setChildRelation instanceof SetOperationRelation) {
+                    sqlBuilder.append(")");
+                }
+            }
+            return sqlBuilder.toString();
+        }
+
+        @Override
+        public String visitTable(TableRelation node, Void outerScope) {
+            StringBuilder sqlBuilder = new StringBuilder();
+            sqlBuilder.append(node.getName());
+
+            if (node.getAliasWithoutNameRewrite() != null) {
+                sqlBuilder.append(" AS ");
+                sqlBuilder.append(node.getAliasWithoutNameRewrite());
+            }
+            return sqlBuilder.toString();
+        }
+
+        @Override
+        public String visitValues(ValuesRelation node, Void scope) {
+            StringBuilder sqlBuilder = new StringBuilder();
+            if (node.getRows().size() == 1) {
+                sqlBuilder.append("SELECT ");
+                List<String> fieldLis = Lists.newArrayList();
+                for (int i = 0; i < node.getRows().get(0).size(); ++i) {
+                    String field = visit(node.getRows().get(0).get(i));
+                    String alias = " AS `" + node.getColumnOutputNames().get(i) + "`";
+                    fieldLis.add(field + alias);
+                }
+
+                sqlBuilder.append(Joiner.on(", ").join(fieldLis));
+            } else {
+                sqlBuilder.append("VALUES(");
+
+                for (int i = 0; i < node.getRows().size(); ++i) {
+                    sqlBuilder.append("(");
+                    List<String> rowStrings =
+                            node.getRows().get(i).stream().map(Expr::toSql).collect(Collectors.toList());
+                    sqlBuilder.append(Joiner.on(", ").join(rowStrings));
+                    sqlBuilder.append(")");
+                }
+                sqlBuilder.append(")");
+            }
+            return sqlBuilder.toString();
+        }
+
+        @Override
+        public String visitTableFunction(TableFunctionRelation node, Void scope) {
+            StringBuilder sqlBuilder = new StringBuilder();
+
+            sqlBuilder.append(node.getFunctionName());
+            sqlBuilder.append("(");
```

Comment:
emm. This class shares a lot of code with SQLLabelBuilder, how about extract some common pattern to base class?

Reply:
ok